gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/InstrumentatedGoogleHadoopFileSystem.java

```diff
@@ -308,4 +316,39 @@ public IOStatistics getIOStatistics() {
   public GhfsInstrumentation getInstrumentation() {
     return this.instrumentation;
   }
+
+  /** Set the Value for http get and head request related statistics keys */
+  private void setHttpStatistics() {
+    try {
+      if (getGcsFs() != null) {
+        try {
+          long get_failures =
+              getGcsFs()
+                  .getGcs()
+                  .getStatistics(GoogleCloudStorageStatistics.ACTION_HTTP_GET_REQUEST_FAILURES)
+                  .longValue();
+          if (get_failures > 0L) {
```

Comment:
should we capture failures for HEAD requests as well ?

Reply:
Updated